pkg/ottl/ottlfuncs/func_duration_to_hours.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type DurationToHoursArguments[K any] struct {
+	Duration ottl.DurationGetter[K] `ottlarg:"0"`
+}
+
+func NewDurationToHoursFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("DurationToHours", &DurationToHoursArguments[K]{}, createDurationToHoursFunction[K])
```

Comment:
@evan-bradley this name ensures we never have a naming conflict in the future, but I'm partial to `Hours` so that we match the Go function it represents.  What do you think?

Reply:
In terms of matching Go function names, I would argue `DurationToHours` is actually closer to how its read when used (`d.Hours()`). That said, I think `Hours` works fine and is a bit shorter, and since we're standardizing on the Go `time` package to represent time values we shouldn't have any conflicts.

Reply:
I changed this in my recent commits.